src/basic/log.c

```diff
@@ -352,7 +352,7 @@ void log_forget_fds(void) {
 }
 
 void log_set_max_level(int level) {
-        assert((level & LOG_PRIMASK) == level);
+        assert(FLAGS_SET(LOG_PRIMASK, level));
```

Comment:
this came up before. it's a bit weird this one. FLAGS_SET() is usually used so that the first arg is some variable and the second arg are constant bits, to check whether the bits are set in the variable. But here the call is weirdly reversed, this ceck if variable bits are set in a fixed bit mask, which i can't wrap my head around even ;-)

So yeah, cocinelle is right of course, the replacement it proposes is correct. But it's also incorrect on a philosophical level if you follow what I mean

so I'd exclude this as a one-of. Or maybe we can tweak coccinelle to skip cases where the second arg is not a constant?

Reply:
Ah, I had a weird feeling I've seen this one before :-)

> so I'd exclude this as a one-of. Or maybe we can tweak coccinelle to skip cases where the second arg is not a constant?

I guess skipping the transformation when the second arg is not a constant makes sense, will do.